packages/ui/src/models/visualization/flows/pipe-visual-entity.ts

```diff
@@ -1,18 +1,30 @@
-import { KameletBinding as KameletBindingModel } from '@kaoto-next/camel-catalog/types';
 import get from 'lodash.get';
 import set from 'lodash.set';
 import { v4 as uuidv4 } from 'uuid';
-import { EntityType } from '../../camel-entities';
-import { KameletBindingStep, KameletBindingSteps } from '../../camel-entities/kamelet-binding-overrides';
+import { EntityType } from '../../camel/entities';
+import { PipeSink, PipeSource, PipeStep, PipeSteps } from '../../camel/entities/pipe-overrides';
 import { BaseVisualCamelEntity, IVisualizationNode, VisualComponentSchema } from '../base-visual-entity';
 import { createVisualizationNode } from '../visualization-node';
 import { KameletSchemaService } from './kamelet-schema.service';
 
-export class KameletBinding implements BaseVisualCamelEntity {
+type PipeFlow = {
+  source: PipeSource;
+  steps: PipeSteps;
+  sink: PipeSink;
+};
+
+export class PipeVisualEntity implements BaseVisualCamelEntity {
   readonly id = uuidv4();
-  type = EntityType.KameletBinding;
+  type = EntityType.Pipe;
+  flow: PipeFlow;
 
-  constructor(public route: Partial<KameletBindingModel> = {}) {}
+  constructor(
```

Comment:
If we follow this path, what would happen with the metadata and other properties from `Pipe`?

If I write the source code as:

```
apiVersion: camel.apache.org/v1alpha1
kind: KameletBinding
metadata:
  name: route-3308
spec:
  source: null
  sink: null
```

wouldn't we lose the rest?

Reply:
**Update:** We don't technically lose the rest of the objects because the `source`, `steps`, and `sink` are linked through a reference on the `constructor` as a side effect.

While this works, it's not ideal because depends on a side effect that might change inadvertently, making the debug process extremely complicated. (Unfortunately, we had many of these situations in the past).

This aligns with point #2 [from my previous comment](https://github.com/KaotoIO/kaoto-next/pull/210#pullrequestreview-1675819446), we should tackle this as soon as we can, otherwise, we would set us up for pain :smiling_face_with_tear: